packages/accounts/src/light-account/account.ts

```diff
@@ -150,6 +152,44 @@ export class LightSmartContractAccount<
     return decodedCallResult;
   }
 
+  encodeUpgradeToAndCall = async (
+    upgradeToImplAddress: Address,
+    upgradeToInitData: Hex
+  ): Promise<Hex> => {
+    const provider = this.rpcProvider;
+    // Step 1: check if the account we're upgrading is in fact a LightAccount
+    const accountAddress = await this.getAddress();
+
+    const storage = await provider.getStorageAt({
+      address: accountAddress,
+      slot: "0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc",
+    });
```

Comment:
wanna leave a comment to humanize the slot and why this is where we look?

Reply:
ah yea lemme do that